staker/stakerapp.go

```diff
@@ -506,6 +506,33 @@ func (app *App) mustSetTxSpentOnBtc(hash *chainhash.Hash) {
 	}
 }
 
+// AddBTCTransactionToDBAndCheckStatus receives a BTC staking transaction that should be
+// already included in the BTC network
+func (app *App) AddBTCTransactionToDBAndCheckStatus(
+	stakerAddr btcutil.Address,
+	stkTx *wire.MsgTx,
+	parsedStakingTx *staking.ParsedV0StakingTx,
+) error {
+	pop, err := app.createPop(stakerAddr)
+	if err != nil {
+		return err
+	}
+
+	if err := app.addTransactionSentToBTC(
+		stkTx,
+		uint32(parsedStakingTx.StakingOutputIdx),
+		parsedStakingTx.OpReturnData.StakingTime,
+		[]*btcec.PublicKey{parsedStakingTx.OpReturnData.FinalityProviderPublicKey.PubKey},
+		babylonPopToDBPop(pop),
+		stakerAddr,
+	); err != nil {
+		return err
+	}
+
+	// not fun to call check transaction, because there is no request and way to get the tx result
+	return app.checkTransactionsStatus()
```

Comment:
Yea, `checkTransactionsStatus`  was not designed to be called in rpc api and its behaviour is undefined if there are other requests flying around 😅 

Two alternatives I see to this:
1. either modify `stakingRequestCmd` and handling of this command, to handle case when transactions is already built as in some sense migrating transaction is special case of post approval flow in which transaction is already built and confirmed on btc chain
2. build all transactions and data in new function `SendPhase1Transaction` and in this function call `WatchStaking` function.  This should use all existing machinery for tracking this transaction and sending it to Babylon chain

Reply:
I used the [waitForStakingTransactionConfirmation](https://github.com/babylonlabs-io/btc-staker/blob/916d27d3df655b59a3c7f4ae005c77e0fabab2fe/staker/stakerapp.go#L563) function to also verify if it is deep enough and complete the existing machinery, let me know what you think about it